status-im/status-react
# Enabling biometry without password during sync

fixes #17928 

### Summary
Currently, when the user syncs device A with device B (mobile), the user is prompted to input the device A password to enable biometry. This was needed because the device A password comes hashed during syncing, while for biometry authentication we need the password in plaintext for the keychain. 

As agreed with @cammellos and @Samyoul, we should be moving away from storing the plaintext password in the keychain in favor of storing it already hashed (we only need it hashed anyway). As such, this PR:
- adds a migration for the keychain password from plaintext to hashed form
- removes the password authentication step for enabling biometry from sync onboarding 

### Testing notes

To test the migration:
1. create an account with enabled biometry (don't use this branch yet)
2. install this PR build
3. try to log in using biometry

If it fails to log in during step 3, then there's something wrong with the migration.

#### Platforms


- Android
- iOS


#### Areas that maybe impacted


##### Functional

- onboarding
- biometry auth

### Steps to test


- start sync process on another device e.g. status-desktop
- open status-mobile
- start sync (by scanning the QR code or entering the sync code)
- when biometry is prompted, enable it and register it

When enabling biometry, the password auth should not appear and the user should be able to log it again with biometry after the app was closed.

status: ready
